sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,171 @@
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
+    fun setPhoneNumber(phoneNumber: String): Klaviyo = apply {
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
     }
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Assigns an external ID to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param id Phone number for active user
+     * @return
+     */
+    fun setExternalId(id: String): Klaviyo = apply {
+        this.setProfile(Profile().setIdentifier(id))
+    }
+
+    /**
+     * Assign arbitrary attributes to the current profile by key
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
      */
-    fun identify(properties: KlaviyoCustomerProperties) {
-        val request = IdentifyRequest(KlaviyoConfig.apiKey, properties)
-        processRequest(request)
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        // TODO debounce so fluent setters don't have to create 1 request per call
```

Comment:
We need this tracked anywhere? Just to make sure we follow up on it

Reply:
Ah thanks, I'd say this is a blocker to even merging this PR, it should have been part of the overall "api updates" work. I will go through my TODOs in android studio.